### PR TITLE
Extended Feedback, Added pre-meeting note

### PR DESCRIPTION
**Pull request by awilli70**
### Alumni

image

image

image

### Student

image

image

image

### CODE REVIEW

File: client/src/components/StudentMentorship.js [hunk 1]
```diff
@@ -365,7 +398,7 @@ class SchedulingCards extends Component {
                         <Button onClick={this.toggleFeedbackModal.bind(this)}>
                             Done
                         </Button>
-                        <Button onClick={this.submitFinalNote.bind(this)} primary>
+                        <Button onClick={this.submitFinalNote.bind(this)} primary disabled={!this.state.publicFeedback || !this.state.privateFeedback}>
```
Comment (awilli70) [inline client/src/components/StudentMentorship.js:401]: Students should not be able to submit without having public or private feedback for the alumni
Testimonials are more special and doesn't have an immediate impact on alumni, and thus should only be submitted when other feedback is given

File: server/routes/requests.js [hunk 2]
```diff
@@ -132,6 +132,7 @@ router.patch('/updateRequest/:id/:timeOffset', passport.authenticate('jwt', {ses
             const dbData = await requestSchema.find({mentor: alumniId, status: status}).populate('student')
             for (let request of dbData) {
                 request.time = await timezoneHelpers.applyTimezone(request.time, timeOffset)
+                request.privateFeedback = undefined;
```
Comment (awilli70) [inline server/routes/requests.js:135]: This removes any evidence of privateFeedback from returned documents

File: client/src/components/AlumniMentorship.js [hunk 3]
```diff
@@ -536,6 +553,30 @@ class RequestCards extends Component {
         })
     }
 
+    toggleAlumniNoteModal(e) {
+        let requestDetails = this.props.requests.find(request => request._id === e.currentTarget.getAttribute('requestid'))
+        console.log(requestDetails)
```
Comment (reshadbinharun) [inline client/src/components/AlumniMentorship.js:558]: Remove debug log here

File: client/src/components/StudentMentorship.js [hunk 4]
```diff
@@ -167,7 +167,9 @@ class SchedulingCards extends Component {
         schedulings: [],
         display: [],
         schedulingDetails: [],
-        feedback: '',
+        publicFeedback: '',
```
Comment (reshadbinharun) [inline client/src/components/StudentMentorship.js:170]: very appropriate field names!

File: client/src/components/StudentMentorship.js (hunk 1 above)
Comment (reshadbinharun) [reply client/src/components/StudentMentorship.js:401]: Makes sense! let's go with it

File: server/routes/requests.js (hunk 2 above)
Comment (reshadbinharun) [reply server/routes/requests.js:135]: Perfect! And we're not losing any of this data on fetches, so that's great

**Pull request merged**